packages/openactive-integration-tests/test/features/leasing/common.js

```diff
@@ -0,0 +1,150 @@
+ *
+ * e.g. `multiplyFetchedOrderItemsIntoBatches(fetchOpportunities, 3)` will return 3 of each OrderItem. This is
+ * equivalent to the real life use case of someone booking an opportunity for themself and 2 friends.
+ *
+ * @param {FetchOpportunitiesFlowStageType} fetchOpportunities
+ * @param {number} multiple
+ */
+function multiplyFetchedOrderItemsIntoBatches(fetchOpportunities, multiple) {
+  const fetchedOrderItems = fetchOpportunities.getOutput().orderItems;
+
+  /** @type {OrderItem[]} */
+  const result = [];
+  for (let i = 0; i < multiple; i += 1) {
+    result.push(
+      // OrderItems are shallow cloned so that we can reset their positions
+      ...fetchedOrderItems.map(orderItem => ({ ...orderItem })),
+    );
+  }
+  // now, reset `position`s so that multiple OrderItems don't have the same positions :o
+  for (let i = 0; i < result.length; i += 1) {
+    result[i].position = i;
+  }
+  return result;
+}
+
+/**
+ * Similar to Common.itForEachOrderItem. Though, this is used for when lease capacity tests
+ * where Orders are made by getting the OrderItemCriteria and creating a batch of N OrderItems for each one
+ * (e.g. 3 of each Order Item Criteria to test the scenario of someone ordering an opportunity for themself
+ * and 2 other friends).
+ *
+ * @param {object} args
+ * @param {OpportunityCriteria[]} args.orderItemCriteriaList
+ * @param {() => ChakramResponse} args.getOrdersApiResponse HTTP response from an Orders API that includes
+ *   OrderItems in the `.orderedItem` field. e.g. C1, C2 or B.
+ * @param {number} args.batchMultiplier How many items are in each batch?
+ * @param {string} testName Label used for it() block when checking OrderItems.
+ * @param {(apiResponseOrderItem: any) => void} cb Callback which runs assertions on OrderItems.
+ */
+function itForEachOrderItemWhereOrderItemsAreBatched({
+  orderItemCriteriaList,
+  getOrdersApiResponse,
+  batchMultiplier,
+}, testName, cb) {
+  const numOrderItems = orderItemCriteriaList.length * batchMultiplier;
+
+  /* This test checks a pre-condition of the subsequent tests for each OrderItem - that the number
+  of OrderItems is balanced with the number of Order Item Criteria */
+  it('Should have the same number of OrderItems as criteria', () => {
+    const apiResponseOrderedItem = getOrdersApiResponse().body.orderedItem;
+    expect(apiResponseOrderedItem).to.be.an('array')
+      .that.has.lengthOf.above(0)
+      .and.has.lengthOf(numOrderItems);
+  });
+  for (let i = 0; i < numOrderItems; i += 1) {
+    it(`OrderItem at position ${i} - ${testName}`, () => {
+      const apiResponseOrderItem = getOrdersApiResponse().body.orderedItem.find(orderItem => (
+        orderItem.position === i
+      ));
+      // eslint-disable-next-line no-unused-expressions
+      expect(apiResponseOrderItem).to.not.be.null
+        .and.to.not.be.undefined;
+      cb(apiResponseOrderItem);
+    });
+  }
+}
```

Comment:
Previously, this was using `Common.itForOrderItem`. This actually missed many of the OrderItems because it only checks an amount of OrderItems which is equal to the number of orderItemCriteria. In these tests, there are more OrderItems than criteria because they duplicate items in order to test things like "booking 3 of the same thing".

This is a modified function which works for this "batched" approach